invertedai/large/initialize.py

```diff
@@ -0,0 +1,265 @@
+        not specifying traffic light state is equivalent to using iai generated light states.
+
+    get_birdview:
+        If True, a birdview image will be returned representing the current world. Note this will significantly
+        impact on the latency.
+
+    get_infractions:
+        If True, infraction metrics will be returned for each agent.
+    
+    random_seed:
+        Controls the stochastic aspects of initialization for reproducibility.
+
+    api_model_version:
+        Optionally specify the version of the model. If None is passed which is by default, the best model will be used.
+
+    display_progress_bar:
+        If True, a bar is displayed showing the progress of all relevant processes.
+    
+    See Also
+    --------
+    :func:`initialize`
+    """
+
+    agent_states_sampled = []
+    agent_attributes_sampled = []
+    agent_rs_sampled = []
+
+    def inside_fov(center: Point, agent_scope_fov: float, point: Point) -> bool:
+        return ((center.x - (agent_scope_fov / 2) < point.x < center.x + (agent_scope_fov / 2)) and
+                (center.y - (agent_scope_fov / 2) < point.y < center.y + (agent_scope_fov / 2)))
+    
+    iterable_regions = None
+    if display_progress_bar:
+        iterable_regions = tenumerate(
+            regions, 
+            total=len(regions),
+            desc=f"Initializing regions"
+        )
+    else:
+        iterable_regions = regions
+
+    for i, region in iterable_regions:
+        region_center = region.center
+        region_fov = region.get_region_fov()
+
+        existing_agent_states, existing_agent_attributes, _ = _get_all_existing_agents_from_regions(regions)
+
+        conditional_agents = list(filter(
+            lambda x: inside_fov(center=region_center, agent_scope_fov=region_fov+AGENT_SCOPE_FOV_BUFFER, point=x[0].center), 
+            zip(existing_agent_states,existing_agent_attributes)
+        ))
+
+        conditional_agent_states = [x[0] for x in conditional_agents]
+        conditional_agent_attributes = [x[1] for x in conditional_agents]
+
+        region_predefined_agent_states = [] if region.agent_states is None else region.agent_states
+        all_agent_states = conditional_agent_states + region_predefined_agent_states
+        all_agent_attributes = conditional_agent_attributes + region.agent_attributes
+
+        try:
+            # Initialize simulation with an API call
+            response = iai.initialize(
+                location=location,
+                states_history=None if len(all_agent_states) == 0 else [all_agent_states],
+                agent_attributes=[] if len(all_agent_attributes) == 0 else all_agent_attributes,
+                get_infractions=get_infractions,
+                traffic_light_state_history=traffic_light_state_history,
+                location_of_interest=(region_center.x, region_center.y),
+                random_seed=random_seed,
+                get_birdview=get_birdview,
```

Comment:
I'm fine with leaving this out for now because it's not-trivial how to interpret this in the "large" version of initialize and drive and there are indeed other visualization methods.